packages/contracts/contracts/L1/OptimismPortal.sol

```diff
@@ -0,0 +1,157 @@
+     * @dev This is intended as a convenience function for EOAs. Contracts should call the
+     * depositTransaction() function directly.
+     */
+    receive() external payable {
+        depositTransaction(msg.sender, msg.value, 30000, false, bytes(""));
+    }
+
+    /**
+     * @notice Finalizes a withdrawal transaction.
+     * @param _nonce Nonce for the provided message.
+     * @param _sender Message sender address on L2.
+     * @param _target Target address on L1.
+     * @param _data Data to send to the target.
+     * @param _gasLimit Gas to be forwarded to the target.
+     * @param _timestamp L2 timestamp of the outputRoot.
+     * @param _outputRootProof Inclusion proof of the withdrawer contracts storage root.
+     * @param _withdrawalProof Inclusion proof for the given withdrawal in the withdrawer contract.
+     */
+    function finalizeWithdrawalTransaction(
+        uint256 _nonce,
+        address _sender,
+        address _target,
+        uint256 _value,
+        uint256 _gasLimit,
+        bytes calldata _data,
+        uint256 _timestamp,
+        WithdrawalVerifier.OutputRootProof calldata _outputRootProof,
+        bytes calldata _withdrawalProof
+    ) external {
+        // Check that the timestamp is 7 days old.
+        unchecked {
+            if (block.timestamp < _timestamp + FINALIZATION_PERIOD) {
+                revert NotYetFinal();
+            }
+        }
+
+        // Get the output root.
+        bytes32 outputRoot = L2_ORACLE.getL2Output(_timestamp);
+
+        // Verify that the output root can be generated with the elements in the proof.
+        if (outputRoot != WithdrawalVerifier._deriveOutputRoot(_outputRootProof)) {
+            revert InvalidOutputRootProof();
+        }
+
+        // Verify that the hash of the withdrawal transaction's arguments are included in the
+        // storage hash of the withdrawer contract.
+        bytes32 withdrawalHash = keccak256(
+            abi.encode(_nonce, _sender, _target, _value, _gasLimit, _data)
+        );
+        if (
+            WithdrawalVerifier._verifyWithdrawalInclusion(
+                withdrawalHash,
+                _outputRootProof.withdrawerStorageRoot,
+                _withdrawalProof
+            ) == false
+        ) {
+            revert InvalidWithdrawalInclusionProof();
+        }
+
+        // Check that this withdrawal has not already been finalized.
+        if (finalizedWithdrawals[withdrawalHash] == true) {
+            revert WithdrawalAlreadyFinalized();
+        }
+
+        l2Sender = _sender;
+        // Make the call.
+        _target.call{ value: _value, gas: _gasLimit }(_data);
+        l2Sender = DEFAULT_L2_SENDER;
+
+        // All withdrawals are immediately finalized. If the ability to replay a transaction is
+        // required, that support can be provided in external contracts.
+        finalizedWithdrawals[withdrawalHash] = true;
+        emit WithdrawalFinalized(withdrawalHash);
+    }
+}
```

Comment:
For symmetry's sake, should we have a `Withdrawals` L1 contract and just make `OptimismPortal` extend both `DepositFeed` and `Withdrawals`?

I would also add a top comment in this file saying that these two facets need to live in the same contracts because we need to withdraw the ETH from the place where it was deposited.